### DISCUSSION TITLE
Personal SSO protected Codespace secrets

### DISCUSSION THREAD

**Discussion by subesokun** [community/community#28974] [Codespaces]
I want to store personal Codespace secrets, and the secrets should be scoped to a particular SSO enabled GH organization to ensure that only Codespaces that are part of that SSO enabled GH organization can access them.

In my GH profile I can create such personal Codespace secrets and assign them to a particular repo that is located within the SSO enabled organization but still the secret is not really protected by SSO. In case my GH account gets hijacked, the intruder can simply update the scope of the secret and allow access from any repo in my GH profile. Then the intruder can simply create a new Codespace in any of my private repos and extract the secrets from it.

Updated idea:

Basically I just want to ensure that a personal secret can't leave nor enter a SSO enabled organization without having an active SSO session. Furthermore, the name of the secret can be already sensitive too and therefore the personal secrets for a certain SSO organizations should  be only shown in the UI if the user has an active SSO session. Probably I'm looking for something like "personalized organization-level secrets". It would be even fine for me if those secrets can't be assigned to multiple organizations if that helps to reduce the complexity.

Initial idea:

_Would it be possible to lock the scope of a secret once it has been created? I think that would solve my issue. Because without performing the SSO into the GH organization, the intruder won’t be able to start a Codespace that has access to the secrets and the scope can't be altered too without recreating the secret. For the sake of security, locking the secrets by default that are bound to a SSO enabled organization would make sense imho._

Or are there any other workarounds to achieve a similar behavior? I know I can create organization specific secrets, but this secret is really specific to my user. E.g. a deploy key that is bound to my particular user and so on.

**Comment by joshaber** [comment]
Hi @subesokun, nothing like that exists currently but it's a great observation! I'll open an internal feature request so we can track it. Thanks!

**Reply by subesokun** [reply]
Thanks a lot for your positive feedback! I was giving this some more thought and probably just locking the secrets will end in a bad UX. Every time you want to update the lists of repos for a secret you'd need to recreate it and there is also no organization-level wildcard available. Basically I just want to ensure that a personal secret can't leave nor enter a SSO enabled organization without having an active SSO session. Furthermore, the name of the secret can be already sensitive too and therefore the personal secrets for a certain SSO organizations should be only shown in the UI if the user has an active SSO session. Probably I'm looking for something like "personalized organization-level secrets". It would be even fine for me if those secrets can't be assigned to multiple organizations if that helps to reduce the complexity.
